### ISSUE TITLE
Waterfox scales opposite to what it should on Wayland

### ISSUE THREAD

**Issue by proairface** [MrAlex94/Waterfox#3514]
### What happened?

I am running OpenSuSe Tumbleweed and Waterfox seems to be the only application that has this bizarre behavior where the scaling is opposite of what it should be, meaning that on my external 1080p screen with 100% scaling in Wayland, it seems as though it is rendering it as 200%+ what I have on my laptop screen. 
Dragging Waterfox to my 4K laptop screen, makes it extremely difficult to see. Text, titlebars, menubars, extentions, everything becomes what I assume to be the 100% that SHOULD be set for my external screen.
So Waterfox scales according to the opposite display it is on it seems, I have never seen this behavior before.
I've tried starting up an X11 session and setting the scaling to 100%, as well as checking if the font size is weird but it is still all default.
Very strange behavior as regular Firefox does not do this, it is only Waterfox and I am at a real headscratcher what this could be. Other apps don't do this either.
I am posting this here as Waterfox is the only app affected by this, so I will for now assume it is an issue with Waterfox, and not my system but I need help further troubleshooting this.
Please let me know what sort of information I can add to this issue to make it easier for the devs if this is my system or waterfox that is acting up.

### Reproducible?

- [X] I have checked that this issue cannot be reproduced on Mozilla Firefox.

### Version

G6

### What platform are you seeing the problem on?

Linux

**Comment by proairface** [comment]
![Screenshot_20240730_202328](https://github.com/user-attachments/assets/99ca72d3-b5cc-4c7c-aaef-1db49c055158)
Here is a screenshot of what I mean. I have to zoom out to 60% to even type this comment. Everything is huge on my external display. I've also tried playing around with the KDE system settings for X11 scaling under the Display section. I've tried to change to let the system, or the apps manage the scaling but nothing seems to affect this issue.

**Comment by proairface** [comment]
![Screenshot_20240730_202726](https://github.com/user-attachments/assets/767927fb-8510-492f-a09c-df01ad91f36e)
For reference, this is what this ticket looks like on my laptop display. I have zoomed in on purpose to make it 1) readable for me to type this comment but 2) to show you the difference in e.g. the navbar on the top and the min/max/close buttons.

**Comment by proairface** [comment]
I figured out what the problem is. I changed the setting "layout.css.devPixelsPerPx" under about:config and that causes some reeeeaaal weird behavior. I have not seen this on GNOME on my old Pop!_OS installation so I think there is a link between OpenSuSe/KDE/Wayland and this setting making it go nuts. Under Pop! I also have a modified PixelsPerPx count but that doesn't cause any issues. The more you know I guess... 
For anyone ever running into this issue again in the future, there you go. Check your about:config settings and reset them just in case to test if this causes any issues. This also explains why Firefox on my OpenSuSe machine was just working correctly, as I barely ever use it and I never bothered to go into the about:config to change the PixelsPerPx value.
